security/keycloak/src/main/java/io/quarkus/ts/security/keycloak/OidcProviderClientResource.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkus.ts.security.keycloak;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+
+import io.quarkus.oidc.AccessTokenCredential;
+import io.quarkus.oidc.OidcProviderClient;
+import io.quarkus.oidc.OidcSession;
+import io.quarkus.oidc.UserInfo;
+
+@Path("/oidc-provider-client")
+public class OidcProviderClientResource {
+
+    @Inject
+    OidcSession oidcSession;
+
+    @Inject
+    OidcProviderClient oidcProviderClient;
+
+    @Inject
+    AccessTokenCredential accessToken;
+
+    @Inject
+    UserInfo userInfo;
+
+    @POST
+    @Path("/token/revoke")
+    public String revokeAccessTokens(String refreshToken) {
+        return oidcSession.logout()
```

Comment:
calling `oidcSession.logout()` may give someone idea that it actually does something for you, but if you drop it and leave just:

```
oidcProviderClient.revokeAccessToken(accessToken.getToken()).await().indefinitely();
oidcProviderClient.revokeRefreshToken(refreshToken).await().indefinitely();
return "Logout, tokens revoked";
```

pass anyway, AFAICT you are using bearer token auth, not code flow.

Reply:
and yes, I read docs and I know you copied it from there so docs suggest that, but that's only relevant when you have a web-app, you are doing a different thing